charts/kong/templates/service-kong-proxy.yaml

```diff
@@ -10,6 +10,7 @@
 {{- include "kong.service" $serviceConfig }}
 {{ if .Values.proxy.ingress.enabled }}
 ---
+{{ if (not (hasKey .Values.proxy.ingress "hosts"))  }}
```

Comment:
Ah, okay, this was added because of https://github.com/Kong/charts/issues/313#issuecomment-802245433

Regardless, that line should no longer be here--it's gone in next, so if SDK usage still breaks with the current next template (https://github.com/Kong/charts/blob/887c5d29e37081e1b33e400750a2b0bb4754304f/charts/kong/templates/service-kong-proxy.yaml#L12-L13), any fix should be targeted to that, not the now-removed legacy handler.